### DISCUSSION TITLE
code questions

### DISCUSSION THREAD

**Discussion by DNXEMPIRE-1** [community/community#82937] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

What is the difference between coding and programming?

**Comment by ghostinhershell** [comment]
Hey there!  👋

Thanks for posting in the GitHub Community, @DNXEMPIRE-1! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've gone ahead and moved this to the correct category for you. Good luck!

**Comment by Esteban4-CRo** [comment]
Simply put, coding is the specific act of translating logic into a machine-readable language (writing the actual lines of code), while programming is the much broader discipline that includes coding but also covers planning, architecture, problem-solving, debugging, testing, and maintaining the entire system—so coding is like laying the bricks, but programming is being the architect who designs the whole building, ensures it won't collapse, and oversees the entire project from start to finish.

**Comment by wgarciar3** [comment]
Coding: The act of writing source code in a programming language (such as C#, Java, Python, or JavaScript). It involves translating a predefined solution into instructions that a computer can execute.

Programming: A much broader process that includes:

  1. Analyzing the problem.
  2. Designing the solution and software architecture.
  3. Choosing algorithms and data structures.
  4. Writing the code (coding).
  5. Testing and debugging.
  6. Optimizing and maintaining the software.

Example

If you are asked to develop a clinic management system:

Programming involves deciding which modules the system will have (patients, appointments, billing, inventory), designing the database, defining the business logic, planning the architecture, writing the code, testing it, and maintaining it.
Coding is simply the process of writing the source code for those modules.

Summary

Coding = Writing code.
Programming = Solving a problem through software; coding is just one part of that process.

Every programmer writes code, but someone can be a coder without being involved in the analysis, design, architecture, or overall software development process.
